pkg/apis/core/validation/validation.go

```diff
@@ -5371,6 +5378,22 @@ func ValidateResourceRequirements(requirements *core.ResourceRequirements, fldPa
 	return allErrs
 }
 
+func validateResourceQuantityHugePageValue(name core.ResourceName, quantity resource.Quantity, opts PodValidationOptions) (bool, error) {
+	if !helper.IsHugePageResourceName(name) {
+		return false, nil
+	}
+
+	if opts.AllowIndivisibleHugePagesValues {
+		return true, nil
```

Comment:
returning early here can lead to accidentally skipping future validation added to this method. prefer this:

```go
if !opts.AllowIndivisibleHugePagesValues && !helper.IsHugePageResourceValueValid(name, quantity) {
  return false, fmt.Errorf("%s is not integer multiple of %s", quantity.String(), name)
}

return nil
```